### PR TITLE
[optimize-WIP](functions) optimize const_column to full convert

### PR DESCRIPTION
**Pull request by zclllyybb**
# Proposed changes
before:
```
mysql [ssb]>select count(from_days(LO_ORDERDATE)) from lineorder_flat;
+----------------------------------+
| count(from_days(`LO_ORDERDATE`)) |
+----------------------------------+
|                                0 |
+----------------------------------+
1 row in set (4.40 sec)
```

after:
```
mysql [ssb]>select count(from_days(LO_ORDERDATE)) from lineorder_flat;
+----------------------------------+
| count(from_days(`LO_ORDERDATE`)) |
+----------------------------------+
|                                0 |
+----------------------------------+
1 row in set (4.21 sec)
```

## Problem summary

make functions with only one argument to be automatically transformed when it's a constant column.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

### CODE REVIEW

File: be/src/vec/functions/function_bitmap.cpp [hunk 1]
```diff
@@ -272,8 +272,7 @@ class FunctionBitmapAlwaysNull : public IFunction {
         auto& null_map = res_null_map->get_data();
         auto& res = res_data_column->get_data();
 
-        ColumnPtr argument_column =
-                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        ColumnPtr argument_column = block.get_by_position(arguments[0]).column;
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap.cpp:275]: use reference

Comment (HappenLee) [inline be/src/vec/functions/function_bitmap.cpp:275]: use reference

File: be/src/vec/functions/functions_comparison.h [hunk 2]
```diff
@@ -112,9 +112,9 @@ struct GenericComparisonImpl {
 
     static void NO_INLINE vector_constant(const IColumn& a, const IColumn& b,
                                           PaddedPODArray<UInt8>& c) {
-        auto b_materialized = b.clone_resized(1)->convert_to_full_column_if_const();
-        for (size_t i = 0, size = a.size(); i < size; ++i)
-            c[i] = Op::apply(a.compare_at(i, 0, *b_materialized, 1), 0);
+        for (size_t i = 0, size = a.size(); i < size; ++i){
```
Comment (HappenLee) [inline be/src/vec/functions/functions_comparison.h:115]: the change seems will cause coredump

File: be/src/vec/functions/function_bitmap.cpp [hunk 3]
```diff
@@ -744,8 +744,7 @@ class FunctionBitmapToArray : public IFunction {
         dest_nested_column = dest_nested_nullable_col->get_nested_column_ptr();
         auto& dest_nested_null_map = dest_nested_nullable_col->get_null_map_column().get_data();
 
-        auto arg_col =
-                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        auto arg_col = block.get_by_position(arguments[0]).column;
```
Comment (HappenLee) [inline be/src/vec/functions/function_bitmap.cpp:747]: use reference

**Comment by zclllyybb** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.52 seconds
 stream load tsv:          474 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         41 seconds loaded 2358488459 Bytes, about 54 MB/s
 stream load orc:          75 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230313103324_clickbench_pr_113542.html

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**